### ISSUE TITLE
Failing to read vector<float> and vector<int> of LZ data using v0.14.3

### ISSUE THREAD

**Issue by 63990** [FAST-HEP/fast-carpenter#89]
Using a conda environment using fast-carpenter v0.14.3.
Trying to read 2 branches:
pulse area: **Br  265 :pulsesODHG.pulseArea_phd : vector<float>**
coincidences: **Br  282 :pulsesODHG.coincidence : vector<int>**        

The processing YAML file is as follows:
```
stages:
    - pulse_cleaning: fast_carpenter.Define
    - coincidence_pulse_area: fast_carpenter.BinnedDataframe

pulse_cleaning:
    variables:
        - coincident_pulse_0: pulsesODHG.coincidence>0
        - pulses_area: pulsesODHG.pulseArea_phd

coincidence_pulse_area:
    binning:
        - {in: pulses_area, bins: {low: 0, high: 10000, nbins: 1000}}
    weights: coincident_pulse_0
```
This fails giving the following output:
```
2019-10-17 09:17:55,384 - alphatwirl.misc.deprecation - WARNING - alphatwirl.misc.deprecation.MultiprocessingDropbox.g(): the option "progressbar" is deprecated. "progressbar=True" is given. use atpbar.disable() instead to turn off progress bars
WARNING:alphatwirl.misc.deprecation:alphatwirl.misc.deprecation.MultiprocessingDropbox.g(): the option "progressbar" is deprecated. "progressbar=True" is given. use atpbar.disable() instead to turn off progress bars
   0.00%                                          |        0 /        1 |:  Background                       Process Worker-1:
   0.00%                                          |        0 /        1 |:  Background                       
Traceback (most recent call last):
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/global/homes/s/seriksen/.local/lib/python3.7/site-packages/mantichora/worker.py", line 27, in run
    self._run_tasks()
  File "/global/homes/s/seriksen/.local/lib/python3.7/site-packages/mantichora/worker.py", line 47, in _run_tasks
    result = task_func()
  File "/global/homes/s/seriksen/.local/lib/python3.7/site-packages/mantichora/main.py", line 18, in _call_
    return self.task(*self.args, **self.kwargs)
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/alphatwirl/concurrently/CommunicationChannel.py", line 16, in _call_
    return self.task(*self.args, **self.kwargs)
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/alphatwirl/loop/EventLoop.py", line 45, in _call_
    self.reader.event(event)
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/alphatwirl/loop/ReaderComposite.py", line 43, in event
    if reader.event(event) is False:
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/fast_carpenter/summary/binned_dataframe.py", line 192, in event
    data = explode(data)
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/fast_carpenter/summary/binned_dataframe.py", line 292, in explode
    return explode(res)
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/fast_carpenter/summary/binned_dataframe.py", line 273, in explode
    lens = pd.DataFrame({col: df[col].str.len() for col in lst_cols})
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/fast_carpenter/summary/binned_dataframe.py", line 273, in <dictcomp>
    lens = pd.DataFrame({col: df[col].str.len() for col in lst_cols})
  File "/global/homes/s/seriksen/.local/lib/python3.7/site-packages/pandas/core/generic.py", line 5175, in _getattr_
    return object.__getattribute__(self, name)
  File "/global/homes/s/seriksen/.local/lib/python3.7/site-packages/pandas/core/accessor.py", line 175, in _get_
    accessor_obj = self._accessor(obj)
  File "/global/homes/s/seriksen/.local/lib/python3.7/site-packages/pandas/core/strings.py", line 1917, in _init_
    self._inferred_dtype = self._validate(data)
  File "/global/homes/s/seriksen/.local/lib/python3.7/site-packages/pandas/core/strings.py", line 1967, in _validate
    raise AttributeError("Can only use .str accessor with string " "values!")
AttributeError: Can only use .str accessor with string values!
^C2019-10-17 11:42:05,129 - atuproot.atuproot_main - WARNING - received KeyboardInterrupt
WARNING:atuproot.atuproot_main:received KeyboardInterrupt
2019-10-17 11:42:05,130 - atuproot.atuproot_main - WARNING - terminating running jobs
WARNING:atuproot.atuproot_main:terminating running jobs
Traceback (most recent call last):
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/bin/fast_carpenter", line 10, in <module>
    sys.exit(main())
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/fast_carpenter/__main__.py", line 77, in main
    _, ret_val = run_carpenter(sequence, datasets, args)
  File "/global/homes/s/seriksen/miniconda3/envs/fast_lz/lib/python3.7/site-packages/fast_carpenter/__main__.py", line 95, in run_carpenter
    ret_val = process.run(datasets, sequence)
  File "/global/homes/s/seriksen/.local/lib/python3.7/site-packages/atuproot/atuproot_main.py", line 66, in run
    return result
UnboundLocalError: local variable 'result' referenced before assignment
```

This setup works fine  on fast-carpenter v0.14.0 but not on any of the future versions (0.14.1 - 0.14.3)

**Comment by benkrikler** [comment]
Hmm, it seems to be triggered only with certain files.  I'm not sure what the difference is yet between the relevant branches in the two files, but I can at least reproduce this.  Will investigate more.

**Comment by benkrikler** [comment]
This should now be fixed as of v0.15.0, so I'm closing this off.
